### PR TITLE
Add waits in ListenerPrimaryTests to mitigate race with List.Add.

### PR DESCRIPTION
**Pull request by cesarbs**
Same as #2038 but for 1.1.

### CODE REVIEW

**Comment by cesarbs** [comment]
Updated with backport of #1230. @Eilon This doesn't involve any product code change and stabilizes tests.

**Comment by cesarbs** [comment]
CI will fail but I verified locally that it builds by reverting 708c4ddcd90ba44a44f64325f65aea7e2c2745c3 (cc @pranavkm).

**Comment by halter73** [comment]
Build seems to be failing because of https://github.com/aspnet/KestrelHttpServer/commit/708c4ddcd90ba44a44f64325f65aea7e2c2745c3#commitcomment-23967740.

File: test/Microsoft.AspNetCore.Server.KestrelTests/ListenerPrimaryTests.cs [hunk 1]
```diff
@@ -189,6 +189,12 @@ public class ListenerPrimaryTests
 
                 await kestrelThreadPrimary.PostAsync(_ => pipe.Dispose(), null);
 
+                // Wait up to 10 seconds for error to be logged
```
Comment (Eilon) [inline test/Microsoft.AspNetCore.Server.KestrelTests/ListenerPrimaryTests.cs]: This looks like 1 second, not 10? 10*100ms == 1sec?

File: test/Microsoft.AspNetCore.Server.KestrelTests/ListenerPrimaryTests.cs [hunk 2]
```diff
@@ -334,6 +346,32 @@ public class ListenerPrimaryTests
 
                 pipe.Dispose();
             }
+	}
+
+        private static async Task AssertResponseEventually(
+            string address,
+            string expected,
+            string[] allowed = null,
+            int maxRetries = 100,
```
Comment (Eilon) [inline test/Microsoft.AspNetCore.Server.KestrelTests/ListenerPrimaryTests.cs]: *This* looks like 10 seconds to me.

**Comment by cesarbs** [comment]
Updated with 30 second timeouts.

**Pull request merged**